core/arch/aarch64/emit_utils.c

```diff
@@ -250,17 +281,17 @@ link_indirect_exit_arch(dcontext_t *dcontext, fragment_t *f,
     else
         exit_target = get_linked_entry(dcontext, target_tag);
 
+    /* Set pc to the last instruction in the stub. */
     pc = (uint *)(stub_pc + exit_stub_size(dcontext, target_tag, f->flags) -
-                  (2 * AARCH64_INSTR_SIZE));
+                  AARCH64_INSTR_SIZE);
 
+    pc = get_stub_branch(pc) - 1;
     /* ldr x1, [x(stolen), #(offs)] */
-    pc[0] = (0xf9400000 | 1 | (dr_reg_stolen - DR_REG_X0) << 5 |
-             get_ibl_entry_tls_offs(dcontext, exit_target) >> 3 << 10);
-    /* br x1 */
-    pc[1] = 0xd61f0000 | 1 << 5;
```

Comment:
Was this a redundant write before?

Reply:
Yes looks like it. The assertion in get_stub_branch should catch the case where it isn't.